staging/src/k8s.io/kubectl/pkg/cmd/cmd.go

```diff
@@ -225,40 +225,25 @@ func (h *DefaultPluginHandler) Execute(executablePath string, cmdArgs, environme
 // HandlePluginCommand receives a pluginHandler and command-line arguments and attempts to find
 // a plugin executable on the PATH that satisfies the given arguments.
 func HandlePluginCommand(pluginHandler PluginHandler, cmdArgs []string) error {
-	var remainingArgs []string // all "non-flag" arguments
-	for _, arg := range cmdArgs {
+	var pluginPath string
+	var pluginArgs []string
+	for i, arg := range cmdArgs {
 		if strings.HasPrefix(arg, "-") {
-			break
+			continue // flag
 		}
-		remainingArgs = append(remainingArgs, strings.Replace(arg, "-", "_", -1))
-	}
-
-	if len(remainingArgs) == 0 {
-		// the length of cmdArgs is at least 1
-		return fmt.Errorf("flags cannot be placed before plugin name: %s", cmdArgs[0])
-	}
-
-	foundBinaryPath := ""
-
-	// attempt to find binary, starting at longest possible name with given cmdArgs
-	for len(remainingArgs) > 0 {
-		path, found := pluginHandler.Lookup(strings.Join(remainingArgs, "-"))
-		if !found {
-			remainingArgs = remainingArgs[:len(remainingArgs)-1]
-			continue
+		for j, part := range cmdArgs[i:] {
+			if strings.HasPrefix(part, "-") {
+				break // flag
+			}
+			possiblePluginName := strings.Join(cmdArgs[i:i+1+j], "-")
+			if path, found := pluginHandler.Lookup(possiblePluginName); found {
```

Comment:
There is a [KEP](https://github.com/kubernetes/enhancements/pull/3639) to enable builtin command shadowing by plugins(for some commands like create). Can this plugin extraction mechanism still be extensible to support that in the future `kubectl --namespace foo create networkpolicy --kubeconfig`?

Reply:
Good question.  I see you have a PR for that touches the same code, so I'm guessing there is some overlap that would be a conflict.

This PR was more an idea to see how it would work and get some thoughts on it because I know people have asked for it.

However, it probably makes sense to get the command shadowing in place first, then do this (if it is something we want to support).

I'll go ahead and close this PR for now.